sqlitecluster/SQLiteNode.cpp

```diff
@@ -1212,10 +1210,15 @@ bool SQLiteNode::update(uint64_t& nextActivity) {
                       << "): consistencyRequired=" << SQLCConsistencyLevelNames[consistencyRequired]
                       << ", commitsSinceCheckpoint=" << _commitsSinceCheckpoint);
             }
+
+            // We return false here *even if* we've finished the command. This lets the caller read any new commands
+            // from the network before we process the next command. This lets any new high-priority commands make their
+            // way into our queue before we process anything else, which may be lower priority.
+            return false;
         }
 
         // If we're the master, see if we're to stand down (and if not, start a new command)
-        if (_state == SQLC_MASTERING && !_currentCommand) {
+        if (_state == SQLC_MASTERING) {
```

Comment:
Before, because we had a condition above that would run only if `_currentCommand` had a ` transaction` that was non-empty, we had this extra condition here as well, which is no longer necessary. Since the above condition has been simplified, and now always returns *false* if hit, we can never reach this block with `_currentCommand` set, we will have returned false in that case.